automation/cluster.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2024 Red Hat, Inc.
+#
+
+set -e
+
+SCRIPT_PATH=$(dirname "$(realpath "$0")")
+
+OCI_BIN=${OCI_BIN:-podman}
+
+OVN_K8S_REPO="https://github.com/ovn-org/ovn-kubernetes.git"
+OVN_K8S_BRANCH="master"
+OVN_K8S_REPO_COMMIT="b4388c5a8766e35d5ae5d63833fd7ee00cf0592f"
+
+OVN_K8S_REPO_PATH="${SCRIPT_PATH}/_ovn_k8s/"
+OVN_K8S_KIND="${SCRIPT_PATH}/_ovn_k8s/contrib"
+
+if [ ! -d ${OVN_K8S_REPO_PATH} ]; then
+    git clone ${OVN_K8S_REPO} --branch ${OVN_K8S_BRANCH} --single-branch  ${OVN_K8S_REPO_PATH}
+    pushd ${OVN_K8S_REPO_PATH}
+        git checkout ${OVN_K8S_REPO_COMMIT}
+    popd
+fi
+
+cluster_up() {
+    (
+        cd "${OVN_K8S_KIND}"
+        ./kind.sh \
+            --experimental-provider ${OCI_BIN} \
+            --num-workers 0 \
```

Comment:
Can the num of nodes be controlled by `export NUM_OF_NODES=?`?

Reply:
No, but it can be controlled by `KIND_NUM_WORKER` (default is 2) `KIND_NUM_MASTER` (default is 1).
I used the `num-workers` flag because setting the env vars directly my introduce side effects.